File: ai_widget/languages/laravel/agent.php
<?php
// Laravel Integration for Craveva AI Agent
// Add to your Laravel controller or service

namespace App\Services;

use Illuminate\Support\Facades\Http;
use Illuminate\Support\Facades\Log;

class CravevaAgentService
{
    protected $agentId;
    protected $apiKey;
    protected $baseUrl;
    
    public function __construct()
    {
        $this->agentId = '69ccc35e7d0ece6ff702487b';
        $this->apiKey = 'cvd_viPEIg-8UsVWKmO7hqfu4LZYxee4AoXE';
        $this->baseUrl = 'https://ai.craveva.com';
    }
    
    /**
     * Send a message to the agent
     *
     * @param string $message
     * @param string|null $userId
     * @param string|null $companyId
     * @param string|null $outletId
     * @return array
     */
    public function chat(string $message, ?string $userId = null, ?string $companyId = null, ?string $outletId = null): array
    {
        $endpoint = $this->baseUrl . '/api/v1/agents/' . $this->agentId . '/chat';
        
        $payload = [
            'message' => $message,
        ];
        
        if ($userId) {
            $payload['user_id'] = $userId;
        }
        if ($companyId) {
            $payload['company_id'] = $companyId;
        }
        if ($outletId) {
            $payload['outlet_id'] = $outletId;
        }
        
        $headers = [
            'Content-Type' => 'application/json',
        ];
        
        if ($this->apiKey) {
            $headers['Authorization'] = 'Bearer ' . $this->apiKey;
        }
        
        try {
            $response = Http::withHeaders($headers)
                ->timeout(30)
                ->post($endpoint, $payload);
            
            if ($response->successful()) {
                return $response->json();
            }
            
            Log::error('Craveva Agent API Error', [
                'status' => $response->status(),
                'body' => $response->body(),
            ]);
            
            return ['error' => 'Failed to get response from agent'];
        } catch (\Exception $e) {
            Log::error('Craveva Agent Exception', [
                'message' => $e->getMessage(),
            ]);
            
            return ['error' => $e->getMessage()];
        }
    }
    
    /**
     * Send a message and return only the output text
     *
     * @param string $message
     * @return string
     */
    public function getResponse(string $message): string
    {
        $response = $this->chat($message);
        return $response['output'] ?? 'Sorry, I could not process that request.';
    }
}

// Usage in a Controller:
// 
// use App\Services\CravevaAgentService;
// 
// public function chat(Request $request)
// {
//     $agent = new CravevaAgentService();
//     $response = $agent->chat($request->input('message'));
//     return response()->json($response);
// }

File: ai_widget/languages/java/agent.java
// Java/Spring Boot Integration for Craveva AI Agent
// Add to your Spring Boot service

package com.yourcompany.service;

import org.springframework.http.*;
import org.springframework.stereotype.Service;
import org.springframework.web.client.RestTemplate;
import org.springframework.web.client.HttpClientErrorException;
import java.util.HashMap;
import java.util.Map;

@Service
public class CravevaAgentService {
    
    private final String agentId = "69ccc35e7d0ece6ff702487b";
    private final String apiKey = "cvd_viPEIg-8UsVWKmO7hqfu4LZYxee4AoXE";
    private final String baseUrl = "https://ai.craveva.com";
    private final RestTemplate restTemplate;
    
    public CravevaAgentService(RestTemplate restTemplate) {
        this.restTemplate = restTemplate;
    }
    
    /**
     * Send a message to the agent
     *
     * @param message The message to send
     * @param userId Optional user ID
     * @param companyId Optional company ID
     * @param outletId Optional outlet ID
     * @return Response from the agent
     */
    public Map<String, Object> chat(String message, String userId, String companyId, String outletId) {
        String endpoint = baseUrl + "/api/v1/agents/" + agentId + "/chat";
        
        HttpHeaders headers = new HttpHeaders();
        headers.setContentType(MediaType.APPLICATION_JSON);
        
        if (apiKey != null && !apiKey.isEmpty()) {
            headers.setBearerAuth(apiKey);
        }
        
        Map<String, Object> payload = new HashMap<>();
        payload.put("message", message);
        
        if (userId != null) payload.put("user_id", userId);
        if (companyId != null) payload.put("company_id", companyId);
        if (outletId != null) payload.put("outlet_id", outletId);
        
        HttpEntity<Map<String, Object>> request = new HttpEntity<>(payload, headers);
        
        try {
            ResponseEntity<Map> response = restTemplate.postForEntity(
                endpoint,
                request,
                Map.class
            );
            
            return response.getBody();
        } catch (HttpClientErrorException e) {
            Map<String, Object> errorResponse = new HashMap<>();
            errorResponse.put("error", e.getMessage());
            errorResponse.put("output", "Sorry, I could not process that request.");
            return errorResponse;
        }
    }
    
    /**
     * Send a message and return only the output text
     *
     * @param message The message to send
     * @return The agent's response text
     */
    public String getResponse(String message) {
        Map<String, Object> response = chat(message, null, null, null);
        return (String) response.getOrDefault("output", "Sorry, I could not process that request.");
    }
}

// Usage in a Controller:
// 
// @RestController
// @RequestMapping("/api/chat")
// public class ChatController {
//     
//     @Autowired
//     private CravevaAgentService agentService;
//     
//     @PostMapping
//     public ResponseEntity<Map<String, Object>> chat(@RequestBody Map<String, String> request) {
//         String message = request.get("message");
//         Map<String, Object> response = agentService.chat(message, null, null, null);
//         return ResponseEntity.ok(response);
//     }
// }
